src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -384,7 +388,22 @@ public override EnvDTE.FileCodeModel GetFileCodeModel(DocumentId documentId)
                 this.EnsureEditableDocuments(changedDocs);
             }
 
-            return base.TryApplyChanges(newSolution, progressTracker);
+            var changedSymbols = RenameSymbolAnnotation
+                .GatherChangedSymbolsInDocumentsAsync(changedDocs, CurrentSolution, currentSolution, CancellationToken.None)
+                .WaitAndGetResult_CanCallOnBackground(CancellationToken.None);
```

Comment:
You don't need the _CanCallOnBackground here at least, since this is running on the UI thread per the initial assert. (Feel free to ignore this if other refactoring shares this with TestWorkspace.)